### DISCUSSION TITLE
Ability to specify how to interpolate between keyframes

### DISCUSSION THREAD

**Discussion by hybridherbst** [needle-tools/custom-timeline-editor#11] [Ideas]
Ideas:
- cubic bezier curve easing as a start
- find a good universal easing function that can be tuned
- vvvv-like Damper (dynamic behaviour)
- ability to vvvv-style "disperse" data changes 

(this is usually a Timeline pain point, as e.g. changing the curves for Vector3 scale is already shitty)

**Comment by marwie** [comment]
Easing functions can be added by implementing ICurveEasing now

They will for now then show up here in the clip context menu (but are already serialized and saved)
![image](https://user-images.githubusercontent.com/5083203/141678475-383fbeff-e93a-4cd5-9506-529211f8e3ec.png)

Use the handle in the timeline to weight the easing

https://user-images.githubusercontent.com/5083203/141678559-4d1787ed-d48a-4cb6-bfdf-5a087fb6e935.mp4

**Comment by hybridherbst** [comment]
Is this easing per keyframe or per track?

**Reply by marwie** [reply]
Per clip (animated field in a track) only for now
